csrc/moe/moe_align_sum_kernels.cu

```diff
@@ -233,15 +233,17 @@ void moe_align_block_size(torch::Tensor topk_ids, int64_t num_experts,
       (num_experts + 1) * sizeof(int32_t);
 
   bool use_global_memory = false;
-  bool use_i16 = false; // Use uint16_t for shared memory token counts
-  if (shared_mem_i16 > device_max_shared_mem) {
-    use_global_memory = true;
-  } else if (shared_mem_i32 > device_max_shared_mem &&
+  bool use_i16 = false;  // Use uint16_t for shared memory token counts
+  if (shared_mem_i32 < device_max_shared_mem) {
+    // use_global_memory = false, use_i16 = false
```

Comment:
Should we just remove this case if it does nothing?

Reply:
I keep it to make the condition tree clearer, but it can be removed as well.

Reply:
I'm ok with leaving this case for clarity but in that case I think we'd want to be explicit
```
  if (shared_mem_i32 < device_max_shared_mem) {
    // Do nothing in this case. We're all set to use int32_t token counts
  } else {
    ...
```

Reply:
> I'm ok with leaving this case for clarity but in that case I think we'd want to be explicit
> 
> ```
>   if (shared_mem_i32 < device_max_shared_mem) {
>     // Do nothing in this case. We're all set to use int32_t token counts
>   } else {
>     ...
> ```

updated